app/src/main/java/com/example/atheneum/activities/AddEditBookActivity.java

```diff
@@ -158,7 +160,34 @@ public void onClick(View v) {
 
     public void scanIsbn(View v) {
         Log.i(TAG, "AddBook*** ISBN scan requested");
-        //TODO
+
+        Intent intent = new Intent(this, ScanBarcodeActivity.class);
+
+        startActivityForResult(intent, 0);
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data){
+        Log.i(TAG, "Return from scan ISBN");
+        if(requestCode == 0){
+            if(resultCode == CommonStatusCodes.SUCCESS){
+                if(data != null){
+                    Barcode barcode = data.getParcelableExtra("Barcode");
+                    isbnEditText.setText(String.valueOf(barcode.displayValue));
```

Comment:
I would make a small change here though, since you already have a valid ISBN you should just auto populate the book info based on the ISBN here instead of relying on the user to do it. @AY-Chen do you still think the auto-populate ISBN button is necessary anymore now that we have scanning working?

Reply:
I've kept the auto-populate button just in case, but I've added auto-populate after scanning.